### DISCUSSION TITLE
Formula fields

### DISCUSSION THREAD

**Discussion by gazflo** [Budibase/budibase#11651] [Help] [answered]
Hi,

I'm looking to create formula fields that extract the latitude and longitude from a Google streetview URL stored in a text field (so I can use them in a map view).

I have successfully created a formula using the following JavaScript:

`let url = $("Streetview");
let arr = url.match(/(@)([^,]*),([^,]*)/);
let num = Number(arr[2]);
return num;`

While this returns the required values, they appear to be displayed as strings.

Thus, when I attempt to filter the values, they do not filter correctly:

![image](https://github.com/Budibase/budibase/assets/12023631/f69118fd-7efe-4f57-b4cc-e77f8e20fed2)

![image](https://github.com/Budibase/budibase/assets/12023631/9f612678-eab5-4f14-9d60-aea242243527)

As per [the instructions](https://docs.budibase.com/docs/formula), I am using JavaScript bindings on a static formula field.

I'm struggling to find a solution and cannot find any documentation regarding static and dynamic formulas. Can someone please help me identify what I'm doing wrong?

**Comment by melohagan** [comment, marked as answer]
Hey @gazflo 

I've updated the docs to include a description of the difference between static/dynamic formulas: https://docs.budibase.com/docs/formula#static-formulas

With that said, there does appear to be an issue with the static formula field filtering if the formula was originally a dynamic formula. I have raised a bug here: https://github.com/Budibase/budibase/issues/11669

I imagine you did initially save the formula as dynamic, and then change it to static later? If so, try deleting the formula column and re-adding it as a new **static** formula.

**Comment by gazflo** [comment]
Thanks @melohagan It was indeed a dynamic formula originally. I have tried replacing with a static formula, but I still experience some issues.  Here are the steps to recreate:

1. Add the new static field using JS (values are blank)
2. Move to another tab (e.g. automations) and back
3. Fields are populated
4. Try filtering -> wrong results

I have recorded a video if it helps: https://youtu.be/WMsscXYYBxs

**Reply by melohagan** [reply]
Hey @gazflo It says that the video is private - I can't view it

**Comment by gazflo** [comment]
My bad. Please try again now.

**Comment by gazflo** [comment]
FYI, the filter doesn't work on the data tab, nor as a dynamic filter in the app.

However, it does work in a view. 

Unfortunately, I need to filter in the app.

**Comment by gazflo** [comment]
Another observation and workaround:

I have two identical pages, both with a table and a dynamic filter.

Page 1 uses a BB table as its dataprovider source. Filtering does not work.
Page 2 uses an unfiltered view of the same BB table as its dataprovider source. Filtering works.

**Reply by melohagan** [reply]
Hey @gazflo 

Thanks for digging into this. I was able to confirm that this is a bug with the static number formulas. 

Would you be able to raise a bug issue? Attaching the youtube video would also be very helpful!

**Comment by gazflo** [comment]
Bug issue raised: Inconsistent filtering of static formula fields #11779

**Reply by melohagan** [reply]
Perfect - thank you very much!
